src/server/command/worldedit/Pattern.zig

```diff
@@ -0,0 +1,53 @@
+const std = @import("std");
+
+const main = @import("main");
+const AliasTable = main.utils.AliasTable;
+const Block = main.blocks.Block;
+const ListUnmanaged = main.ListUnmanaged;
+const NeverFailingAllocator = main.heap.NeverFailingAllocator;
+
+blocks: AliasTable(Entry),
+
+const Entry = struct {
+	block: Block,
+	chance: f32,
+};
+
+pub fn initFromString(allocator: NeverFailingAllocator, source: []const u8) !@This() {
+	var specifiers = std.mem.splitScalar(u8, source, ',');
+	var totalWeight: f32 = 0;
+
+	var weightedEntries: ListUnmanaged(struct {block: Block, weight: f32}) = .{};
+	defer weightedEntries.deinit(main.stackAllocator);
+
+	while(specifiers.next()) |specifier| {
+		var iterator = std.mem.splitScalar(u8, specifier, '%');
+
+		// This code first assumes that specifier has form `weight%addon:block` and only if parsing of weight fails it tries to parse it as `addon:block`.
```

Comment:
Multiple colons are allowed when specifying the block data value, this comment makes it seem like they are not.
The code is correct though.

Reply:
This comment explains what, not why, so it can be removed anyway.